lib/systems/inspect.nix

```diff
@@ -114,4 +116,22 @@ rec {
   platformPatterns = mapAttrs (_: p: { parsed = {}; } // p) {
     isStatic = { isStatic = true; };
   };
+};
+
+warnEfiDeprecated = throw "(U)EFI support is a feature of specific motherboards rather than entire architectures or platforms; this predicate has been deprecated because its name is misleading";
+
+in
+result // {
+  patterns = result.patterns // {
+    isEfi = warnEfiDeprecated result.patterns.isEfi;
+  };
+
+  # Unfortunately a strictness bug in lib.systems.equals means that
+  # we can't lib.warn on accesses to boolean values.  If
+  # https://github.com/NixOS/nixpkgs/pull/238331 is merged we can go
+  # back to using Nix `==` and this warning can be enabled.
+  #
+  #predicates = result.predicates // {
+  #  isEfi = argument: warnEfiDeprecated (result.predicates.isEfi argument);
+  #};
```

Comment:
This problem was introduced by

- https://github.com/NixOS/nixpkgs/pull/237512

After that PR, we are no longer able to deprecate boolean-valued attributes of platform objects because `lib.systems.equals` is too strict.

We could get back to using `==`, which is lazy enough to handle this correctly, by merging

- https://github.com/NixOS/nixpkgs/pull/238331